esp/esp/program/modules/handlers/mapgenmodule.py

```diff
@@ -78,17 +78,14 @@ def usermap(self, request, tl, one, two, module, extra, prog):
             context['num_users'] = users.count()
 
             #   Summarize users by state and zip code
-            states_raw = ESPUser.objects.filter(id__in=[user.id for user in users]).select_related('contactinfo_set', 'contactinfo_set__address_state'
-                ).values('contactinfo__address_state'
-                ).order_by('contactinfo__address_state'
-                ).annotate(count = Count('contactinfo__address_state'))
-            states = {state['contactinfo__address_state']: state['count'] for state in states_raw if state['contactinfo__address_state']}
-
-            zipcodes_raw = ESPUser.objects.filter(id__in=[user.id for user in users]).select_related('contactinfo_set', 'contactinfo_set__address_zip'
-                ).values('contactinfo__address_zip'
-                ).order_by('contactinfo__address_zip'
-                ).annotate(count = Count('contactinfo__address_zip'))
-            zipcodes = {zipcode['contactinfo__address_zip']: zipcode['count'] for zipcode in zipcodes_raw if zipcode['contactinfo__address_zip']}
+            #   Only get contact infos that are for the actual users (not guardians or emergency contacts)
+            states_raw = ContactInfo.objects.filter(user__in=users, as_user__isnull=False
```

Comment:
I don't quite understand this as_user condition.  How does it tell us that the ContactInfo is for the actual user?  I thought as_user was related to registration profiles.

Reply:
as_user is the set of registration profiles that use the contact info as the `contact_user` relationship. Otherwise, they use them for `contact_emergency` or `contact_guardian` relationship (if the user is a student). If we find that the contact info has been used as the `contact_user` relationship, then we know this is the contact info for the actual user.